firestore/save_data_batch_writes.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package firestore
+
+// [START firestore_data_batch_writes]
+
+import (
+	"context"
+	"log"
+
+	"cloud.google.com/go/firestore"
+)
+
+func batchWrite(ctx context.Context, client *firestore.Client) error {
```

Comment:
issue: I think that we'll want to construct `Context` objects and the `Client` inside of the snippet. For the client, in particular, we want to demonstrate how to close the client.

Reply:
I'm unwilling to expand scope of this PR to include net new code that isn't focused on ensuring code inside region tags will compile.

I'm happy to undo the region tag expansion, we can keep the sample move in place so later change to region tags won't require coordinating docs update. Would that be preferrable to leaving this as-is?

Reply:
No, what you have is fine. I think it's fair to separate refactoring the snippet from a simple move & region tag expansion. Thanks!